analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnValueIgnored.cs

```diff
@@ -61,41 +60,39 @@ protected override void Initialize(SonarAnalysisContext context)
 
         private static void CheckExpressionForPureMethod(SyntaxNodeAnalysisContext context, ExpressionSyntax expression)
         {
-            if (!(expression is InvocationExpressionSyntax invocation))
+            if (expression is not InvocationExpressionSyntax invocation)
```

Comment:
I think this if statement can be merged with the next one (line 68)

Reply:
I inverted the 2 returns De-Morgan-style and merged them with the correct case to have a single if block, good catch.